client/verta/verta/client.py

```diff
@@ -509,26 +511,55 @@ def set_project(
         self._ctx = _Context(self._conn, self._conf)
         self._ctx.workspace_name = workspace
 
+        return self._get_or_create_project(
+            self._conn,
+            self._conf,
+            self._ctx,
+            name=name,
+            desc=desc,
+            tags=tags,
+            attrs=attrs,
+            public_within_org=public_within_org,
+            visibility=visibility,
+            id=id,
+        )
+
+    @staticmethod
+    def _get_or_create_project(
```

Comment:
This whole file's changes are basically factoring out static versions of these `get_or_create` methods so that they can be called without needing to instantiate a new `Client`. See the absolute nightmare that is `Client.__init__()`—this company's oldest extant Python function—for why we want to avoid that.

Ideally these would actually be moved to new classes—which I have vague, vague plans for— but I wanted to keep this diff as readable as possible for now.